openshift-storage-libs/openshiftstoragelibs/baseclass.py

```diff
@@ -1206,3 +1214,77 @@ def validate_glusterfsd_memory_usage(self, size_limit):
                     "Failed memory used  of glusterfsd {} is greater than the"
                     " expected size {} for node/pod {}".format(
                         memory_used, size_limit, gluster_node))
+
+    def heketi_volumes_in_batch(
+            self, heketi_client_node, heketi_server_url, vol_count, size=1,
+            batch_amount=8, timeout=300, interval=5):
+        """Creates heketi volume in batches.
+
+        Args:
+            heketi_client_node (str): Node on which cmd has to be executed.
+            heketi_server_url (str): Heketi server url.
+            vol_count (int): Count of volumes's to be create.
+            size (str): Volume size.
+            batch_amount (int): Amount of POD's to be create in one batch.
+            timeout (int): timeout for one batch.
+            wait_step ( int): wait step.
+
+        Returns:
+            list: list of heketi volumes.
+        """
+        h_vols, async_obj = [], []
+
+        # Fetch heketi volume list before operation
+        h_vols_count_initial = len(heketi_volume_list(
+            heketi_client_node, heketi_server_url, json=True).get('volumes'))
+
+        # Temporary replace g.run with g.async_run in heketi_volume_create func
+        # to be able to run it in background.Also, avoid parsing the output as
+        # it won't be json at that moment. Parse it after reading the async
+        # operation results.
+        def run_async(cmd, hostname, raise_on_error=True):
+            async_op = g.run_async(host=hostname, command=cmd)
+            async_obj.append(async_op)
+            return async_op
+
+        while vol_count > 0:
+            # Change batch_amount if pvc_count < batch_amount
+            if vol_count < batch_amount:
+                batch_amount = vol_count
+            for _ in range(batch_amount):
+                with mock.patch.object(
+                        json, 'loads', side_effect=(lambda j: j)):
+                    with mock.patch.object(
+                            command, 'cmd_run', side_effect=run_async):
+                        heketi_volume_create(
+                            heketi_client_node, heketi_server_url,
+                            size, json=True)
+            h_vols_count_initial += batch_amount
+
+            # Wait for all the volume to be created
+            for w in Waiter(timeout=timeout, interval=interval):
```

Comment:
This will wait for given interval for each batch not for all volumes. Add variable to reduce some time interval say 600 sec from interval passed from function.

Reply:
Fixed